src/main/java/com/google/cloud/pubsublite/kafka/SharedBehavior.java

```diff
@@ -18,17 +18,24 @@
 
 import static com.google.cloud.pubsublite.kafka.KafkaExceptionUtils.toKafka;
 
+import com.google.cloud.pubsublite.AdminClient;
 import com.google.cloud.pubsublite.Partition;
 import com.google.cloud.pubsublite.TopicPath;
 import com.google.common.collect.ImmutableList;
+import java.time.Duration;
 import java.util.List;
+import java.util.concurrent.TimeUnit;
 import org.apache.kafka.common.PartitionInfo;
 
 /** Shared behavior for producer and consumer. */
 final class SharedBehavior {
-  private SharedBehavior() {}
+  AdminClient client;
```

Comment:
This needs to have close() called on it. If you want to do this, you need to make SharedBehavior implement AutoCloseable.

Reply:
Ah, right. Done. I forgot about Closeable, thanks!